src/app/notebook/notebook-markdown.component.ts

```diff
@@ -0,0 +1,160 @@
+import { DOCUMENT } from '@angular/common';
+import { Component, Inject, Input, OnChanges, SecurityContext } from '@angular/core';
+import { DomSanitizer, SafeHtml } from '@angular/platform-browser';
+import { MarkdownWrapperService } from '../shared/markdown-wrapper/markdown-wrapper.service';
+import { join, replaceAll, selectBestFromMimeBundle } from './notebook-helpers';
+
+@Component({
+  selector: 'app-notebook-markdown',
+  templateUrl: './notebook-markdown.component.html',
+})
+export class NotebookMarkdownComponent implements OnChanges {
+  @Input() cell: any;
+  @Input() baseUrl: string;
+  html: string | SafeHtml = '';
+  private domParser = new DOMParser();
+
+  constructor(
+    private markdownWrapperService: MarkdownWrapperService,
+    private sanitizer: DomSanitizer,
+    @Inject(DOCUMENT) private document: Document
+  ) {}
+
+  ngOnChanges(): void {
+    // BE VERY CAREFUL when modifying this function, because to accomodate MathJax markup,
+    // which is destroyed by the Angular sanitizer, it must implement its own sanitization scheme.
+    const dangerousHtml = this.compileMarkdown(join(this.cell.source), this.cell.attachments);
+
+    // Sanitize using both the markdown wrapper and Angular sanitizers.
+    const sanitizedHtml = this.sanitizer.sanitize(SecurityContext.HTML, this.markdownWrapperService.customSanitize(dangerousHtml));
+
+    // Format embedded TeX math expressions.
+    const mathjaxedSanitizedHtml = this.formatMath(sanitizedHtml);
+
+    // Mark as pre-sanitized and assign to html field.
+    this.html = this.sanitizer.bypassSecurityTrustHtml(mathjaxedSanitizedHtml);
+  }
+
+  compileMarkdown(markdown: string, attachments: any): string {
+    // Adjustment the markdown to support attached notebook images and embedded TeX math expressions.
+    let adjusted = markdown;
+    adjusted = this.convertBackslashedDollars(adjusted);
+    adjusted = this.inlineAttachedImages(adjusted, attachments);
+    adjusted = this.preprocessLatexMath(adjusted);
+    // Convert the adjusted markdown to HTML using our wrapper.
+    return this.markdownWrapperService.customCompile(adjusted, this.baseUrl);
+  }
+
+  formatMath(html: string): string {
+    // Format embedded math using MathJax: https://docs.mathjax.org/en/latest/web/typeset.html
+    // Find the global MathJax object.
+    const mathjax = (<any>this.document?.defaultView)?.MathJax;
+    if (mathjax == undefined) {
+      console.log('mathjax not found');
+      return html;
+    }
+
+    // Convert the input HTML to a DOM representation.
+    // DOMParser creates nodes that are detached from the browser context,
+    // which should prevent any embedded code or handlers from firing.
+    // For example, an `img` tag with an `onerror` handler.
+    const doc = this.domParser.parseFromString(html, 'text/html');
```

Comment:
The input to `parseFromString` is the output from the sanitizers, which probably is always well-formed HTML.  Apparently, `parseFromString` always returns a valid DOM, but if there's an error, it's a `parseerror` element.  So, just in case, I will add a check, which displays nothing if the result is a `parseerror` element.

Reply:
Decided to return a resanitized value on parser error.